drivers/interrupt_controller/intc_nxp_gint.c

```diff
@@ -0,0 +1,313 @@
+	}
+
+	LOG_DBG("Enabled port %d pin %d with polarity %d", port, pin, polarity);
+
+	return 0;
+}
+
+int nxp_gint_disable_pin(const struct device *dev, uint8_t port, uint8_t pin)
+{
+	uint32_t reg;
+	const struct nxp_gint_config *config = dev->config;
+	struct nxp_gint_data *data = dev->data;
+
+	/* Validate port number */
+	if (port >= GINT_PORT_COUNT) {
+		LOG_ERR("Invalid port number: %d (max: %d)", port, GINT_PORT_COUNT - 1);
+		return -EINVAL;
+	}
+
+	/* Validate pin number */
+	if (pin > 31) {
+		LOG_ERR("Invalid pin number: %d (max: 31)", pin);
+		return -EINVAL;
+	}
+
+	/* Disable pin */
+	K_SPINLOCK(&data->lock) {
+		reg = gint_read_reg(config, GINT_PORT_ENAN_OFFSET(port));
+		reg &= ~BIT(pin);
+		gint_write_reg(config, GINT_PORT_ENAN_OFFSET(port), reg);
+	}
+
+	LOG_DBG("Disabled port %d pin %d", port, pin);
+
+	return 0;
+}
+
+bool nxp_gint_is_pending(const struct device *dev)
+{
+	uint32_t ctrl;
+
+	if (!dev) {
+		return false;
+	}
+
+	const struct nxp_gint_config *config = dev->config;
+
+	/* Read control register and check interrupt bit */
+	ctrl = gint_read_reg(config, GINT_CTRL_OFFSET);
+	return (ctrl & GINT_CTRL_INT_BIT) != 0;
+}
+
+int nxp_gint_clear_pending(const struct device *dev)
+{
+	uint32_t ctrl;
+
+	if (!dev) {
+		return -EINVAL;
+	}
+
+	const struct nxp_gint_config *config = dev->config;
+
+	/* Clear interrupt flag, the flag can be cleared by writing 1 to it */
+	ctrl = gint_read_reg(config, GINT_CTRL_OFFSET);
+	if (0U != (ctrl & GINT_CTRL_INT_BIT)) {
+		gint_write_reg(config, GINT_CTRL_OFFSET, ctrl);
+	}
+
+	LOG_DBG("Cleared pending interrupt");
+
+	return 0;
+}
+
+int nxp_gint_register_callback(const struct device *dev,
+			       nxp_gint_callback_t callback,
+			       void *user_data)
+{
+	if (!dev) {
+		return -EINVAL;
+	}
+
+	struct nxp_gint_data *data = dev->data;
+
+	data->callback = callback;
+	data->user_data = user_data;
+
+	LOG_DBG("Registered callback");
+
+	return 0;
+}
+
+static void nxp_gint_reset_reg(const struct device *dev)
+{
+	const struct nxp_gint_config *config = dev->config;
+
+	for (int port = 0; port < GINT_PORT_COUNT; port++) {
+		gint_write_reg(config, GINT_PORT_ENAN_OFFSET(port), 0U);
+	}
+
+	gint_write_reg(config, GINT_CTRL_OFFSET, GINT_CTRL_INT_BIT);
+}
+
+static int nxp_gint_init(const struct device *dev)
+{
+	const struct nxp_gint_config *config = dev->config;
+	struct nxp_gint_data *data = dev->data;
+
+	/* Initialize data structure */
+	data->callback = NULL;
+	data->user_data = NULL;
+
+	/* Release the GINT controller from reset */
+	(void)reset_line_deassert_dt(&config->reset);
+
+	/* Clear the registers */
+	nxp_gint_reset_reg(dev);
+
+	/* Init the interrupt */
+	config->irq_init_func();
+
+	return 0;
+}
+
+#define NXP_GINT_IRQ_CONFIG(n)							\
+	static void nxp_gint_irq_init_##n(void)					\
+	{									\
+		IRQ_CONNECT(DT_INST_IRQN(n),					\
+			    DT_INST_IRQ(n, priority),				\
+			    nxp_gint_isr,					\
+			    DEVICE_DT_INST_GET(n),				\
+			    0);							\
```

Comment:
your indent is wrong

Reply:
I guess this is because github shows tab size as 4 characters, I set tab size as 8 in my local env, the result is right.